### PR TITLE
Redis post

### PR DESCRIPTION
**Pull request by icanb**

### CODE REVIEW

File: _drafts/making-backend-scalable.markdown [hunk 1]
```diff
@@ -7,11 +7,11 @@ tags: [tango, redis, scalability]
...
+[Tango](https://github.com/autolab/Tango) is a stand-alone, RESTful service that Autolab uses as the back-end for autograding. Tango receives grading jobs from Autolab's front-end, adds them in a job queue, assigns them to available containers for grading and shepherds the job throuout the process. In its early days, Tango was mostly used for jobs that ran in under 5 seconds. However, over the recent semesters, Autolab has grown to host classes like Distributed Systems, Machine Learning. Some of these classes have assessments that are computationally intensive on large datasets and contains bigger files. As we looked into how to hand larger loads, we were running into the problem of how to manage queued jobs and distribute them to different instances in our back-end. To this end, we decided to take the initial step for turning Tango into a distributed system by implementing a persistent memory using Redis and using the producer consumer model.
```
Comment (jez) [inline _drafts/making-backend-scalable.markdown]: - adds them **to** a job queue
  - preposition usage
- shepherds the job **through** the process
  - Typo
- Autolab has grown to host classes like Distributed Systems, Machine Learning, **and Storage Systems--classes with significantly higher compute requirements.**
  - Lists of three sound better, and joining the two sentences (either with a dash or a semicolon) will improve readability

Comment (jez) [inline _drafts/making-backend-scalable.markdown]: In other news, this is a nice intro now.

File: _drafts/making-backend-scalable.markdown [hunk 2]
```diff
@@ -26,28 +26,27 @@ Therefore we decided to improve our architecture by switching to multi-process m
 
 ## Persistent Memory Model using Redis ##
 
-The in-memory job queue was the barrier to making the system robust. We want to store jobs on an independent system and make it *persistent*
+The in-memory job queue was the barrier to making the system robust. We want to store jobs on an independent system and make it *persistent*.
 
-![Tango with Persistent Memory Model]({{site-url}}/assets/redis2.png)
-_Diagram 2: Tango with Persistent Memory Model_
+![Tango with Persistent Memory Model]({{ site-url }}/assets/redis2.png)
```
Comment (jez) [inline _drafts/making-backend-scalable.markdown]: Spaces vs no spaces between the `{{...}}` is fine, but the hyphen needs to be a dot: `site.url` is a YAML entry called `url` that's accessible from `_config.yml`, so it gets namespaced in the post to `site.`

File: _drafts/making-backend-scalable.markdown [hunk 3]
```diff
@@ -26,28 +26,27 @@ Therefore we decided to improve our architecture by switching to multi-process m
...
 
-We are planning to add modules to support the other options in the future. More details about this will be give in the next sub-section.
 
 ### Implementation ###
 
-While making the switch, we wanted to keep the ability to run Tango without dependency on the external service in order to keep the backwards competibility and ease of development environment setup.
+While making the switch, we wanted to keep the ability to run Tango without a dependency on the external service in order to keep backwards competibility and simplify setting up a local development environment.
```
Comment (jez) [inline _drafts/making-backend-scalable.markdown]: - competibility -> compatibility
  - spelling

**Comment by jez** [comment]
There's nowhere to comment this, but the images still don't have a transparent background. From #1:
- If (as Mihir mentioned) these diagrams really were made with Google Draw, it'd be awesome if you could Download as SVG from the "File" menu.
  - SVGs can have transparent backgrounds
  - SVG take up less space and achieve higher resolutions with no effort
  - it looks like you just screenshot'ed these (redis1.png below shows the red, dotted underline from a misspelled word)

**Comment by jez** [comment]
This is probably good enough once those changes are made.

File: _drafts/making-backend-scalable.markdown (hunk 2 above)
Comment (icanb) [inline _drafts/making-backend-scalable.markdown]: site.url doesn't seem to work.

**Pull request merged**